dags/mlflow-dag.py

```diff
@@ -0,0 +1,239 @@
+        df['occupation'] = df['occupation'].apply(lambda x: 'Unknown' if x == '?' else x)
+        df['native_country'] = df['native_country'].apply(lambda x: 'Unknown' if x == '?' else x)
+
+
+        # Drop Extra/Unused Columns
+        df.drop(columns=['education_num', 'relationship', 'functional_weight'], inplace=True)
+
+        return df
+
+    @task
+    def feature_engineering(df: pd.DataFrame):
+        """Feature engineering step
+        
+        Returns pandas dataframe via XCom to GCS bucket.
+
+        Keyword arguments:
+        df -- data from previous step pulled from BigQuery to be processed. 
+        """
+
+        
+        # Onehot encoding 
+        df = pd.get_dummies(df, prefix='workclass', columns=['workclass'])
+        df = pd.get_dummies(df, prefix='education', columns=['education'])
+        df = pd.get_dummies(df, prefix='occupation', columns=['occupation'])
+        df = pd.get_dummies(df, prefix='race', columns=['race'])
+        df = pd.get_dummies(df, prefix='sex', columns=['sex'])
+        df = pd.get_dummies(df, prefix='income_bracket', columns=['income_bracket'])
+        df = pd.get_dummies(df, prefix='native_country', columns=['native_country'])
+
+
+        # Bin Ages
+        df['age_bins'] = pd.cut(x=df['age'], bins=[16,29,39,49,59,100], labels=[1, 2, 3, 4, 5])
+
+
+        # Dependent Variable
+        df['never_married'] = df['marital_status'].apply(lambda x: 1 if x == 'Never-married' else 0) 
+
+
+        # Drop redundant colulmn
+        df.drop(columns=['income_bracket_<=50K', 'marital_status', 'age'], inplace=True)
+
+        return df
+
+
+    @task.python()
+    def grid_search_cv(df: pd.DataFrame, **kwargs):
+        """Train and validate model
+        
+        Returns accuracy score via XCom to GCS bucket.
+
+        Keyword arguments:
+        df -- data from previous step pulled from BigQuery to be processed. 
+        """
+
+        
+        y = df['never_married']
+        X = df.drop(columns=['never_married'])
+
+
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=55, stratify=y)
+        train_set = lgb.Dataset(X_train, label=y_train)
+        test_set = lgb.Dataset(X_test, label=y_test)
+
+        params = {'objective': 'binary', 'metric': ['auc', 'binary_logloss']}
+
+        grid_params = {
+            'learning_rate': [0.01, .05, .1], 
+            'n_estimators': [50, 100, 150],
+            'num_leaves': [31, 40, 80],
+            'max_depth': [16, 24, 31, 40],
+            'boosting_type': ['gbdt'], 
+            'objective': ['binary'],
+            'seed': [55],
+            }
+
+        model = lgb.LGBMClassifier(**params)
+        grid_search = GridSearchCV(model, param_grid=grid_params, verbose=1, cv=5, n_jobs=-1)
+
+        mlflow.set_tracking_uri('http://host.docker.internal:5000')
+        try:
+            # Creating an experiment 
+            mlflow.create_experiment('census_prediction')
+        except:
```

Comment:
You could add the specific exception you'd expect if the experiment already exists and output a log message saying as much.